src/core/memory.c

```diff
@@ -104,3 +123,47 @@ void *allocate_huge_page(int size)
   return NULL;
 }
 
+// SIGSEGV handler: attempts map packet memory on demand
+void allocate_on_sigsegv(int sig, siginfo_t *si, void *unused)
+{
+  uint64_t address = (uint64_t)si->si_addr;
+  if ((address & 0x500000000000ULL) != 0x500000000000ULL) {
+    // This is not DMA memory: die.
+    //exit(139);
+  } else {
+    uint64_t physaddr = address  & ~0x500000000000ULL;
+    uint64_t physpage = physaddr & ~(2*1024*1024-1);
+    uint64_t virtpage = address  & ~(2*1024*1024-1);
```

Comment:
There is a bug here inherited from my hacky demand mapping:

Huge page size is hard-coded to 2MB but this is not safe and we should use the value detected in `memory.lua`.